homeassistant/components/modbus/binary_sensor.py

```diff
@@ -99,18 +102,33 @@ def device_class(self) -> Optional[str]:
         """Return the device class of the sensor."""
         return self._device_class
 
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
     def update(self):
         """Update the state of the sensor."""
-        if self._input_type == INPUT_TYPE_COIL:
-            result = self._hub.read_coils(self._slave, self._address, 1)
-        else:
-            result = self._hub.read_discrete_inputs(self._slave, self._address, 1)
+
+        def _set_unavailable():
+            if self._available:
+                _LOGGER.error(
+                    "No response from hub %s, slave %s, address %s",
+                    self._hub.name,
+                    self._slave,
+                    self._address,
+                )
+            self._available = False
+
         try:
+            if self._input_type == INPUT_TYPE_COIL:
+                result = self._hub.read_coils(self._slave, self._address, 1)
+            else:
+                result = self._hub.read_discrete_inputs(self._slave, self._address, 1)
+            if isinstance(result, (ModbusException, ExceptionResponse)):
+                _set_unavailable()
+                return
             self._value = result.bits[0]
-        except AttributeError:
-            _LOGGER.error(
-                "No response from hub %s, slave %s, address %s",
-                self._hub.name,
-                self._slave,
-                self._address,
-            )
+            self._available = True
+        except ConnectionException:
```

Comment:
We should only wrap the code in `try…except` that can raise. So that should only be the `result =`  code.

Reply:
Fixed.